### PR TITLE
Implement extra buildings field component

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Implement extra buildings field component
- Add basic Switch component (checkbox).
- Update onSubmit prop to accept async callback function.
- Update label for extra building input.

[Screen Recording 2022-09-27 at 09.56.03.mov](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/OgXegTwxM9IeuXHZyw5h/4886c887-b69f-4b49-a4ca-70ca2f9e87bd/Screen%20Recording%202022-09-27%20at%2009.56.03.mov)

## Justify why they are needed

There's no design for how to display added extra buildings so I just came up with something minimal.

The Switch is not complete/animated but I don't consider that the most critical to work on now.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #597** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/597" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/597?utm_source=stack-comment).

File: apps/store/src/components/PriceForm/ExtraBuildingsField.tsx [hunk 1]
```diff
@@ -1,60 +1,143 @@
...
+      }),
+    })
   }
 
   return (
-    <Dialog.Root>
-      <Dialog.Trigger asChild>
-        <Button type="button">{translateLabel(field.label)}</Button>
-      </Dialog.Trigger>
+    <Dialog.Root open={isOpen} onOpenChange={() => setIsOpen((prev) => !prev)}>
```
Comment (alebedev) [inline apps/store/src/components/PriceForm/ExtraBuildingsField.tsx:52]: Nit: Isn't this too much jsx for single component?

**Review by alebedev — APPROVED** [review]
Looks good, could use some structure at markup level

**Review by robinandeer — COMMENTED** [review]
@alebedev curious why you think what "too much jsx" is? Why? :)

But I agree that this component is a little too complex so we could try to break it up.

However, I would do that when we get updated designs

File: apps/store/src/components/PriceForm/ExtraBuildingsField.tsx (hunk 1 above)
Comment (gustaveen) [reply apps/store/src/components/PriceForm/ExtraBuildingsField.tsx:52]: As in the file is becoming too big? As you suggested Robin, we could split it up into smaller components when we have finalised design

File: apps/store/src/components/PriceForm/ExtraBuildingsField.tsx [hunk 2]
```diff
@@ -71,3 +154,40 @@ const convertExtraBuilding = (data: Record<string, FormDataEntryValue>): ExtraBu
...
+  alignItems: 'center',
+}))
+
+const MutedText = styled.p(({ theme }) => ({
+  fontFamily: theme.fonts.body,
+  fontSize: theme.fontSizes[1],
+  color: theme.colors.gray700,
+}))
+
+const Sup = styled.sup({
```
Comment (gustaveen) [inline apps/store/src/components/PriceForm/ExtraBuildingsField.tsx:189]: `sup`, that's a first for me :)

**Review by gustaveen — APPROVED** [review]
Good job! 🎉

nit: add components to Storybook

**Pull request merged**